salrun/Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM centos:7
+
+LABEL name="LSST SAL Runtime Dependencies" \
+    vendor="LSST" \
+    license="GPLv3"
+
+RUN yum -y install epel-release
+RUN yum -y install python36 python36-devel python36-pip \
+    git make patch wget \
+    gcc-c++
```

Comment:
These are not runtime dependencies. Including them in every container is excessive. Note that my versions of the Dockerfiles enabled you to create separate -build containers that had the required development tools; you've stripped that out of most of these.

Reply:
These were added to get later builds working. For example, the headerservice and  also atarchive need `gcc-c++` to install further down dependancies. For example `fitsio` needs `gcc-c++` and `patch` to build. And the headerservice needs `wget` to build.

Reply:
But then those should be added directly to the build parts of those containers, not to the underlying runtime.

Reply:
At the moment `gcc-c++` is needed by _all_ containers except for `salobj` so I don't believe it makes sense to make the build more complex by the small saving is size of excluding `gcc-c++`. Moreover, we are still using these images for development and we want to make the useful. We could revise these once we have a system that we have ready for production.

Reply:
Multi-stage builds allow you to have both: development containers and production deployment containers.  And the discipline of not having development tools in deployed production containers can be useful.  (Debugging tools could be included, although there are also ways of debugging using "sidecar" containers.)

I give up for now, though; this may be technical debt we have to clean up later, as you mention.